lib/livebook/runtime/attached.ex

```diff
@@ -42,12 +43,23 @@ defmodule Livebook.Runtime.Attached do
             node_manager_opts: [parent_node: node(), capture_orphan_logs: false]
           )
 
-        {:ok, %{runtime | server_pid: server_pid}}
+        {:ok, %{runtime | node: node, server_pid: server_pid}}
 
       :pang ->
         {:error, "node #{inspect(node)} is unreachable"}
     end
   end
+
+  # We need to append the hostname late because
+  # net_kernel has not yet started during config.
+  defp append_hostname(node) do
+    with :nomatch <- :string.find(Atom.to_string(node), "@"),
+         <<suffix::binary>> <- :string.find(Atom.to_string(:net_kernel.nodename()), "@") do
```

Comment:
@josevalim curious: is there a reason to use `:net_kernel.nodename()` over `node()`?

Reply:
OTP 24 adds the ability of setting --name/--sname of the current node to `undefined`, so one is randomly picked. In those cases, `node()` does not return the appropriate value, but the above does: https://github.com/elixir-lang/elixir/commit/729882b8f66907ebdc913e1433f9b937d344d762

In our case it doesn't matter, because it is about the external node, but it is good to be consistent!

Reply:
Perfect, thanks!